### ISSUE TITLE
`termux-api` commands are no-ops.

### ISSUE THREAD

**Issue by empjustine** [termux/termux-packages#361]
`$ termux-dialog` does nothing besides hanging terminal.

```
$ tail /data/data/com.termux/files/usr/bin/termux-dialog
done
shift $(($OPTIND-1))

if [ $# != 0 ]; then echo "$SCRIPTNAME: too many arguments"; exit 1; fi

set -- $ARG_M $ARG_P
if [ -n "$ARG_I" ]; then set -- "$@" $ARG_I "$OPT_I"; fi
if [ -n "$ARG_T" ]; then set -- "$@" $ARG_T "$OPT_T"; fi

/data/data/com.termux/files/usr/libexec/termux-api Dialog "$@"  
```

```
$ logcat
D/AndroidRuntime( 5385): >>>>>> START com.android.internal.os.RuntimeInit uid 10150 <<<<<<
D/AndroidRuntime( 5385): CheckJNI is OFF
I/art     ( 5385): jiangtaottt OpenDexFilesFromOat open_oat_file.get() == nullptr /system/framework/am.jar
I/art     ( 5385): jiangtaottt OpenDexFilesFromOat oat_location == nullptr /system/framework/am.jar
E/System  ( 5385): elements.add : /system/framework/am.jar  /system/framework/am.jar   /system/framework/am.jar
D/AndroidRuntime( 5385): Calling main entry com.android.commands.am.Am
D/AndroidRuntime( 5385): Shutting down VM
```

```
$ ls -la /system/framework/am.jar 
-rw-r--r--    1 root     root         23071 Apr  1 06:25 /system/framework/am.jar
```

**Comment by empjustine** [comment]
As reported on https://github.com/termux/termux-packages/issues/334#issuecomment-260204997,

I found the issue. On my firmware (I think this is a Lenovo issue), random applications are denied from starting from not-launcher intents. I had to disable "Restrict to launch" and it worked.
![](https://cdn.discordapp.com/attachments/155490004666155008/247436118692200468/Screenshot_2016-11-13-17-00-29.png)

**Comment by Panadestein** [comment]
Hi. First of all I will like to thank the developer(s) for giving us the power of linux in our pockets. Concerning this issue, still not working for me in Android 6.0. I have followed all the steps mentioned here https://termux.com/add-on-api.html, and additionally granted permissions to Termux:API. No luck, just void output all time. I have also checked the output of logcat -s termux:* termux-api:* and nothing. I'm not so clear of how it works, the only thing that I have installed until now in zsh, python and his friends (numpy, matplotlib etc), and it works perfectly.
